src/jit/codegenlinear.cpp

```diff
@@ -18,6 +18,92 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "emit.h"
 #include "codegen.h"
 
+/*
+*   Iniatilize spill tracking logic and mark register holding arguments at the beggining of the fucntion (first block)
+*/
+void CodeGen::initializeRegisterSet()
+{
+    unsigned   varNum;
+    LclVarDsc* varDsc;
+
+    /* Initialize the spill tracking logic */
+
+    regSet.rsSpillBeg();
+
+    /* If any arguments live in registers, mark those regs as such */
+
+    for (varNum = 0, varDsc = compiler->lvaTable; varNum < compiler->lvaCount; varNum++, varDsc++)
+    {
+        /* Is this variable a parameter assigned to a register? */
+
+        if (!varDsc->lvIsParam || !varDsc->lvRegister)
+        {
+            continue;
+        }
+
+        /* Is the argument live on entry to the method? */
+
+        if (!VarSetOps::IsMember(compiler, compiler->fgFirstBB->bbLiveIn, varDsc->lvVarIndex))
+        {
+            continue;
+        }
+
+        /* Is this a floating-point argument? */
+
+        if (varDsc->IsFloatRegType())
+        {
+            continue;
+        }
+
+        noway_assert(!varTypeIsFloating(varDsc->TypeGet()));
+
+        /* Mark the register as holding the variable */
+
+        assert(varDsc->lvRegNum != REG_STK);
+        if (!varDsc->lvAddrExposed)
+        {
+            regSet.verifyRegUsed(varDsc->lvRegNum);
+        }
+    }
+}
+
+/*
+*   Initialize Scopes, register, gcInfo and current life variables structures
+*   used in the generation of blocks' code before.
+*/
+void CodeGen::initializeStructuresBeforeBlockCodeGeneration()
```

Comment:
I might simplify this name to simply `initialize`.

Reply:
I am changing to "initialize".

It initializes a lot of structures but, does it sounds quiet generic?